### PR TITLE
Add dirty form checking to user settings

### LINKED ISSUE #182: Change in profile settings is forgotten when tab is changed
## Description
When doing change in Profile settings and moving to another tab without saving changes the change is forgotten.

## Steps to reproduce
1. Login
2. Open Settings
3. Change anything in Profile
4. Switch to OTP tab

## Expected output
Dialog to ask if I want to save the changes.

## Current output
Change is silently forgotten.

### PR DESCRIPTION
**Pull request by ryanlerch**
This adds dirty form checking to user profile settings,
and user keys settings. If a user makes a change to these
forms, then tries to leave the page, a dialog is presented
alerting them to the fact they have unsaved changes.

Resolves: #182

Signed-off-by: Ryan Lerch <rlerch@redhat.com>

### CODE REVIEW

File: noggin/templates/_form_macros.html [hunk 1]
```diff
@@ -39,3 +39,8 @@
     </div>
   {% endif %}
 {% endmacro %}
+
+{% macro unsaved_changes() %}
+<script src="{{ url_for('static', filename='js/vendor/jquery.dirty-0.7.2/jquery.dirty.js') }}"></script>
+<script>$("form").dirty({preventLeaving: true})</script>
```
Comment (abompard) [inline noggin/templates/_form_macros.html:45]: You may want to wrap that in `$(...)` to make sure it's executed after the document has loaded, no?

File: noggin/templates/user-settings-profile.html [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 {% import '_form_macros.html' as macros %}
 {% block settings_content %}
-<form class="needs-validation" action="{{ url_for('user_settings_profile', username=user.username) }}" method="post" novalidate>
+<form class="needs-validation" id="profile-form" action="{{ url_for('user_settings_profile', username=user.username) }}" method="post" novalidate>
```
Comment (abompard) [inline noggin/templates/user-settings-profile.html]: Why set the id here?

File: noggin/templates/_form_macros.html (hunk 1 above)
Comment (abompard) [reply noggin/templates/_form_macros.html:45]: Also, I'm thinking we may want that on all our forms, don't we? If so we can do it at the top-level template.

File: noggin/templates/user-settings-profile.html (hunk 2 above)
Comment (ryanlerch) [reply noggin/templates/user-settings-profile.html]: ha! was originally selecting by that ID, then stopped. this was just a leftover...

File: noggin/templates/_form_macros.html (hunk 1 above)
Comment (ryanlerch) [reply noggin/templates/_form_macros.html:45]: There are some that we def dont want this on, like login, or register...

**Review by abompard — APPROVED** [review]
Thanks!

**Pull request merged**